internal/compiler-bridge/src/main/scala/xsbt/Dependency.scala

```diff
@@ -126,18 +126,42 @@ final class Dependency(val global: CallbackGlobal) extends LocateClassFile with
             // The dependency comes from a class file
             binaryDependency(pf.file, binaryClassName)
           case _ =>
-          // TODO: If this happens, scala internals have changed. Log error.
+            reporter.error(
+              NoPosition,
+              s"Internal error: ${binaryClassName} comes from unknown origin ${at}"
+            )
         }
       }
 
-      val onSource = dep.to.sourceFile
+      val targetSymbol = dep.to
+      val onSource = targetSymbol.sourceFile
       if (onSource == null) {
-        // Dependency is external -- source is undefined
-        classFile(dep.to) match {
-          case Some((at, binaryClassName)) =>
-            processExternalDependency(binaryClassName, at)
-          case None =>
-            debuglog(Feedback.noOriginFileForExternalSymbol(dep.to))
+        // Ignore packages right away as they don't map to a class file/jar
+        if (targetSymbol.hasFlag(scala.tools.nsc.symtab.Flags.PACKAGE)) None
+        // Ignore `Any` which by default has no `associatedFile`
+        else if (targetSymbol == definitions.AnyClass) ()
+        else {
+          classFile(targetSymbol) match {
+            case Some((at, binaryClassName)) =>
+              // Associated file is set, so we know which classpath entry it came from
+              processExternalDependency(binaryClassName, at)
+            case None =>
+              /* If there is no associated file, it's likely the compiler didn't set it correctly.
```

Comment:
In the general case, we're covering some error, but we don't know which error :smile: We've been lucky and we've found one of these errors by surfacing the bug with default arguments. I've not opened a bug report because it's not clear how to reproduce, but I've given steps to Lukas and pinged him on the bug ticket that this PR fixes so that he can clarify the situation. We'll move from there.
  
The issue here is that fixing this instance of errors is not trivial and have to be fixed bug per bug because of the way the class file parser works.